docs/contracts/errors.rst

```diff
@@ -0,0 +1,58 @@
+.. index:: ! error
+
+.. _errors:
+
+******
+Errors
+******
+
+Errors in Solidity provide a convenient and gas-efficient way to explain to the
+user why an operation failed. They can be defined inside and outside of contracts.
+
+::
+
+    // SPDX-License-Identifier: GPL-3.0
+    pragma solidity >=0.7.0 <0.9.0;
```

Comment:
I think we should add an example of an error defined outside of a contract to this snippet for completeness.